teps/0093-tkn-cli-extension-for-signverify.md

```diff
@@ -0,0 +1,347 @@
+feature has become "implemented", major changes should get new TEPs.
+
+The canonical place for the latest set of instructions (and the likely source
+of this file) is [here](/teps/NNNN-TEP-template/README.md).
+
+-->
+
+<!--
+This is the title of your TEP.  Keep it short, simple, and descriptive.  A good
+title can help communicate what the TEP is and should be considered as part of
+any review.
+-->
+
+<!--
+A table of contents is helpful for quickly jumping to sections of a TEP and for
+highlighting any additional information provided beyond the standard TEP
+template.
+
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [Implementation Pull request(s)](#implementation-pull-request-s)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+There is a growing open-source ecosystem around tekton [catalog](https://github.com/tektoncd/catalog) and [artifactHUB](https://artifacthub.io), where number of tasks/pipelines are made available ready-to-use in our CI/CD. These tasks provides commonly used functions like `git-clone`, `curl`, etc. and also includes vendor specific functions like `aws-cli`, `openshift-client`. Such a marketplace is essential in practicing DRY principle. But, as we are building such an open-distribution framework, we need to ensure task distribution is secure. 
+
+Objective in this proposal is to faciliate an easy and handy approach for task/pipeline providers to sign their artifacts and for consumers to verify them using native `tkn cli` commands.
+
+We have an open-source reference implementation available at: [tap8stry-pipelines](https://github.com/tap8stry/tapestry-pipelines).
+
+Also, this complements another great [TEP-0091](https://github.com/tektoncd/community/blob/43643c958fe9cdc76b847996a37d83836656e8eb/teps/0091-verified-task-bundles.md). In TEP-0091, the proposal is to allow sign verification and enforcement at the controller. While in this proposal, we are suggesting an CLI extension for developers to sign and verify pipelines locally at composition. Also, allow sign/verification for plain YAMLs as well as bundles.
```

Comment:
I think it makes sense to have CLI support for [TEP-0091](https://github.com/tektoncd/community/blob/43643c958fe9cdc76b847996a37d83836656e8eb/teps/0091-verified-task-bundles.md) - im wondering if it would make sense to wait to see the details of that design before implementing this? i.e. what if what is implemented for this doesn't match what that TEP ends up needing

the [other TEP-0091 about in-toto layout creation and verification in the CLI](https://github.com/tektoncd/community/pull/534) seems like it is treading similar ground as well?